Archipelago.HollowKnight/ArchipelagoRandomizer.cs

```diff
@@ -140,23 +141,22 @@ public void Randomize()
             }
 
             // Scout all locations
-            void ScoutCallback(LocationInfoPacket packet)
+            var locations = new List<long>(session.Locations.AllLocations);
+            Task<LocationInfoPacket> packetTask = session.Locations.ScoutLocationsAsync(locations.ToArray());
+
+            LocationInfoPacket packet = packetTask.Result;
```

Comment:
couple notes on this:

1. Nothing about this is actually waiting for the scout task to finish
2. Moving this into the BeginInvoke won't help, the point of that is that the stuff in that block *does run on the main thread* because unity doesn't like threading.

Unless I'm missing something it probably should be something like `ScoutLocationsAsync(...).ContinueWith((packetTask) => ThreadSupport.BeginInvoke(<do stuff with packetTask.result here>)`